src/api/v2/types.rs

```diff
@@ -237,29 +246,32 @@ impl Reply for SubscriptionId {
 pub enum RequestType {
 	Version,
 	Status,
+	Submit,
 }
 
 #[derive(Deserialize)]
 pub struct Request {
 	#[serde(rename = "type")]
 	pub request_type: RequestType,
 	pub request_id: String,
```

Comment:
It is a good idea to use [`uuid::Uuiid`](https://docs.rs/uuid/1.4.1/uuid/struct.Uuid.html) here instead of string also. Generally, it is better to avoid strings as much as possible in types.